saleor/payment/utils.py

```diff
@@ -143,15 +140,12 @@ def create_payment(
         "billing_postal_code": billing_address.postal_code,
         "billing_country_code": billing_address.country.code,
         "billing_country_area": billing_address.country_area,
-        "currency": currency,
-        "gateway": gateway,
-        "total": total,
-        "return_url": return_url,
-        "psp_reference": external_reference or "",
     }
 
-    payment, _ = Payment.objects.get_or_create(defaults=defaults, **data)
-    return payment
+    if payment_token:
+        return Payment.objects.get_or_create(token=payment_token, defaults=data)[0]
+    else:
```

Comment:
`else` is not needed here. You're returning in `if` condition

Reply:
I don't think I get it. This function has previously always called `get_or_create`. This caused a second 50% payment to be ignored as it was treated to be identical with the first one.

I decided to keep this function somewhat backwards compatible and update an existing payment by the token, while in most cases a new one will be created (the "else").

Reply:
Maybe I wasn't clear enough. 
```
if payment_token:
    return Payment(...)
return Payment(...)
```
`else` is not needed as you have `return` inside an `if` condition. Just small cosmetic change